### DISCUSSION TITLE
I want to refetch data every 5s from useContractReead, please help me ˆˆ

### DISCUSSION THREAD

**Discussion by minhhoang2404** [wevm/wagmi#2732] [Q&A]
const { data: Short, isSuccess: isSuccessShort }: any = useContractRead({
    address: VAULT_ADDRESS,
    abi: VAULT_ABI,
    enabled: isConnected && timeInterval,
    functionName: 'getPosition',
    args: [address, USDC_ADDRESS, WBTC_ADDRESS, false],
  });

**Comment by SaadNiaziDev** [comment]
You can use watch 
https://wagmi.sh/react/hooks/useContractRead#watch-optional

Or 
(This method do refresh but if reload page after 3 or 4 , it will again wait for 5 sec)

`const { data: Short, refetch, isSuccess: isSuccessShort }:any = useContractRead();`

`useEffect(()=>{`
`const interval = setInterval(()=>{`
`refetch();`
`},5000};`

`return()=> clearInterval(interval)`

**Reply by minhhoang2404** [reply]
yes, thank bro ˆˆ

**Comment by dost21** [comment]
yes ser
